### PR TITLE
Reserve ENI used by hostclasses during VPC creation

### PR DESCRIPTION
**Pull request by jichen-amplify**
Some AWS resources (Elasticache, RDS, Lambda, EC2 instances, etc) can allocate ENIs in a VPC. This causes a problem if a EC2 instance spins up later that is configured to use that static IP. This mostly happens during CI spinup when everything is spinning up at the same time and its possible for a resource to grab a ENI before a EC2 instance gets it.

This PR intends to fix this by changing the VPC spinup process to create ENIs preemptively before spinning up any other resources. This should cause AWS to use some other IP address when spinning up EC, RDS, etc.

### CODE REVIEW

File: bin/disco_vpc_ui.py [hunk 1]
```diff
@@ -28,6 +28,9 @@ def parse_arguments():
                                help='What to call the new environment.')
     parser_create.add_argument('--type', dest='vpc_type', required=True,
                                help='What type of environment to create (as defined in config).')
+    parser_create.add_argument('--skip-enis', dest='skip_enis', action='store_const',
```
Comment (jichen-amplify) [inline bin/disco_vpc_ui.py]: By default, ENIs with static IP addresses would be automatically pre-allocated. This flag is to allow for stopping this behavior.

Another option is to reverse the behaviors between the default and the flag. But doing so would require changing the CI spin job to include the new flag.

Comment (jconstance-amplify) [inline bin/disco_vpc_ui.py]: Why would we want to stop this behavior? Sounds like we're fixing a race condition / bug here, no?

Comment (jichen-amplify) [inline bin/disco_vpc_ui.py]: I was just thinking in the case of creating sandboxes, not all hostclasses with static ENIs need to be spun up. That's why I added the option here. At the same time, there's not much cost to allocate these ENIs either, so I am ok either way.

**Comment by jichen-amplify** [comment]
build

**Pull request merged**